src/submission/migration/migration-manager.ts

```diff
@@ -471,9 +471,9 @@ export namespace MigrationManager {
         if (result && result.length > 0) {
           // if invalid mark as invalid and update document metadata
           if (!dryRun) {
-            const updatedDonor = await updateStatsForInvalidDonorToBe(donor, result);
+            const updatedDonor = await calcDonorCoreEntityStats(_.cloneDeep(donor) as Donor, {});
```

Comment:
`calcDonorCoreEntityStats` clones the first argument in the function already. this is double cloning.

Reply:
Thanks I meant to review the use of cloning, will update

Reply:
Aha the issue here is DeepReadonly<Donor>
So I could update this line to `const updatedDonor = await calcDonorCoreEntityStats(donor as Donor, {});`

Reply:
OK something about the original way this line is written passes our unit tests.
When I try to remove this cloning, the tests fail with error : `TypeError: Cannot assign to read only property 'primaryDiagnosis' of object '#<Object>'`

Certain functions are using DeepReadonly<Donor> which we cannot modify inside the stat-calculation function
But double cloning until the DeepRead goes away is also wrong
Will have to look into this

Reply:
I updated the handling for cloning + types, had to leverage a trick from StackOverflow but it seems to be working. Let me know if this looks good to you.